src/app/FakeLib/UnitTest/Expecto/Expecto.fs

```diff
@@ -0,0 +1,91 @@
+[<AutoOpen>]
+/// Contains tasks to run [xUnit](https://github.com/haf/expecto) v2 unit tests.
+module Fake.Testing.Expecto
+
+open System
+open System.IO
+open System.Text
+open Fake
+
+/// CLI parameters available if you use Tests.runTestsInAssembly defaultConfig argv in your code:
+type ExpectoParams =
+    {
+      /// Extra verbose output for your tests
+      Debug : bool
+      /// Run all tests in parallel. Default is true
+      Parallel : bool
+      /// Filter a specific hierarchy to run
+      Filter : string
+      /// Filter a specific test case to run.
+      FilterTestCase : string
+      /// Filter a specific test list to run.
+      FilterTestList : string
+      /// Run only provided tests.
+      Run : string list
+      /// Doesn't run tests, print out list of tests instead.
+      ListTests : bool
+      /// Working directory
+      WorkingDirectory : string
+    }
+    override this.ToString() =
+        let append (s: string) (sb: StringBuilder) = sb.Append s
+        let appendIfTrue p s sb = 
+            if p then append s sb else sb
+        let appendIfNotNullOrWhiteSpace p s (sb: StringBuilder) = 
+            if String.IsNullOrWhiteSpace p |> not 
+            then sprintf "%s%s " s p |> sb.Append
+            else sb
+        let appendList list s (sb: StringBuilder) = 
+            let filtered = list |> List.filter (String.IsNullOrWhiteSpace >> not)
+            if List.isEmpty filtered then sb
+            else
+                filtered |> String.concat " " |> sprintf "%s%s " s |> sb.Append
+        StringBuilder()
+        |> appendIfTrue this.Debug "--debug "
+        |> appendIfTrue this.Parallel "--parallel "
+        |> appendIfTrue (not this.Parallel) "--sequential "
+        |> appendIfNotNullOrWhiteSpace this.Filter "--filter "
+        |> appendIfNotNullOrWhiteSpace this.FilterTestCase "--filter-test-case "
+        |> appendIfNotNullOrWhiteSpace this.FilterTestList "--filter-test-list "
+        |> appendList this.Run "--run "
+        |> toText
+
+    static member defaultParams =
+        {
+            Debug = false
+            Parallel = true
+            Filter = ""
+            FilterTestCase = ""
+            FilterTestList = ""
+            Run = []
+            ListTests = false
+            WorkingDirectory = ""
+        }
+
+let Expecto (setParams : ExpectoParams -> ExpectoParams) (assemblies : string seq) =
+    let args = setParams ExpectoParams.defaultParams
+    let argsString = string args
+    let runAssembly testAssembly =
```

Comment:
Please trace the taskstart like in https://github.com/fsharp/FAKE/blob/master/src/app/FakeLib/DocFxHelper.fs#L45